### DISCUSSION TITLE
Feature Request: Support multiple column UNPIVOT in simplified syntax

### DISCUSSION THREAD

**Discussion by Alex-Monahan** [duckdb/duckdb#6981] [Ideas]
Howdy @Mytherin (and All)!
 
While documenting the UNPIVOT statement, I came across a situation where it looks like the simplified unpivot syntax can't fully match the functionality of the SQL Standard syntax. I may be mistaken! I couldn't find an exact test for this scenario in the main PR, so I can't tell if this is a bug or a new feature request.

The main reason for this is just to give folks an alternative to the SQL standard syntax for all unpivot scenarios. The SQL standard method does work!

I can also document that this is unsupported if it is not easy or desirable to add - not a problem!

```sql
--Example data from test suite
CREATE OR REPLACE TABLE monthly_sales(empid INT, dept TEXT, Jan INT, Feb INT, Mar INT, April INT);
INSERT INTO monthly_sales VALUES
(1, 'electronics', 100, 200, 300, 100),
(2, 'clothes', 100, 300, 150, 200),
(3, 'cars', 200, 400, 100, 50);
```

| empid |    dept     | Jan | Feb | Mar | April |
|-------|-------------|-----|-----|-----|-------|
| 1     | electronics | 100 | 200 | 300 | 100   |
| 2     | clothes     | 100 | 300 | 150 | 200   |
| 3     | cars        | 200 | 400 | 100 | 50    |

SQL Standard multi-column unpivot:
```sql
SELECT 
  empid, 
  dept,
  month, 
  sales_jan_feb, 
  sales_mar_apr 
FROM monthly_sales 
UNPIVOT(
  (sales_jan_feb, sales_mar_apr) 
  FOR month IN (
    (jan, feb), (mar, april)
  )
);
```

| empid |    dept     |   month   | sales_jan_feb | sales_mar_apr |
|-------|-------------|-----------|---------------|---------------|
| 1     | electronics | Jan_Feb   | 100           | 200           |
| 1     | electronics | Mar_April | 300           | 100           |
| 2     | clothes     | Jan_Feb   | 100           | 300           |
| 2     | clothes     | Mar_April | 150           | 200           |
| 3     | cars        | Jan_Feb   | 200           | 400           |
| 3     | cars        | Mar_April | 100           | 50            |

Various attempts to replicate with simplified syntax:
```sql
UNPIVOT monthly_sales ON (jan, feb), (mar, april) INTO NAME month VALUE sales_jan_feb, sales_mar_april;
Error: Parser Error: syntax error at or near ","
LINE 1: ...il) INTO NAME month VALUE sales_jan_feb, sales_mar_april;
                                                  ^

UNPIVOT monthly_sales ON ((jan, feb), (mar, april));
Error: Parser Error: PIVOT IN list must contain columns or lists of columns

UNPIVOT monthly_sales ON (jan, feb), (mar, april);
Error: Binder Error: UNPIVOT name count mismatch - got 1 names but 2 expressions

UNPIVOT monthly_sales ON (jan, feb), (mar, april) INTO NAME jan_feb, mar_april;
Error: Parser Error: syntax error at or near ","
LINE 1: ...n, feb), (mar, april) INTO NAME jan_feb, mar_april;
                                                  ^

UNPIVOT monthly_sales ON (jan, feb), (mar, april) INTO NAME month VALUE (sales_jan_feb, sales_mar_april);
Error: Parser Error: syntax error at or near "("
LINE 1: ...b), (mar, april) INTO NAME month VALUE (sales_jan_feb, sales_mar_april);
                                                  ^
```

**Comment by Mytherin** [comment]
I actually found this as well yesterday and addressed it in #6961, see [this commit](https://github.com/duckdb/duckdb/pull/6961/commits/d666b709f3bada67ca33b624c2d7e85374cee4f9).

**Reply by Alex-Monahan** [reply]
Will grab the latest CLI! Thanks as always and sorry for missing that one!
